HW1/P6/P6.py

```diff
@@ -0,0 +1,72 @@
+import findspark
+findspark.init()
+import pyspark
+import time
+import re
+from random import random
+from bisect import bisect
+start = time.time()
+sc = pyspark.SparkContext(appName='test')
+
+def mapFunc(word):
+	wordList = []
+	word = word.encode('utf-8')
+	wordList = re.split(" ", word)
+	return wordList
+
+
+def filterFunc(word):
+	if len(word) == 0:
+		return False
+	return (not word[:-1].isupper()) if word[-1] == '.' else  not word.isupper() and not word.isdigit()
+
+textRDD = sc.textFile('pg100.txt',8)
+rawRDD = textRDD.flatMap(mapFunc)
+wordsRDD = rawRDD.filter(filterFunc) # Now wordsRDD contains all the desired words
+
+firstWords, secondWords, thirdWords = wordsRDD.zipWithIndex().map(lambda (k, v): (v, k)), wordsRDD.zipWithIndex().map(lambda (k, v):(v-1,k)).filter(lambda (k, v): k >= 0), wordsRDD.zipWithIndex().map(lambda (k, v): (v-2, k)).filter(lambda (k, v):k >= 0)
+threeWordSet = firstWords.join(secondWords).join(thirdWords).map(lambda (k, v): v)
```

Comment:
I don't think you partitioned your RDD's. The join will likely have an extremely high cost consequently. You can use "partitionBy" to avoid that.